libs/libc/stdlib/lib_cxa_atexit.c

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_cxa_atexit.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <nuttx/atexit.h>
+#include <nuttx/compiler.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+FAR void *__dso_handle weak_data;
+FAR void *__dso_handle = &__dso_handle;
```

Comment:
Shouldn't it be
```suggestion
FAR void *__dso_handle[] weak_data;
FAR void *__dso_handle = (void *)&__dso_handle;
```
?

Reply:
I think the way I declare it is correct, cannot find reference where it is declared the way you do it

Reply:
I mean is `__dso_handle` intended to be provided by linker? If yes, then I'm trying to understand what the code means
```
FAR void *__dso_handle weak_data;
FAR void *__dso_handle = &__dso_handle;
```
`__dso_handle` will hold `NULL` while `FAR void *__dso_handle = &__dso_handle;` -- I need to go and read some books...

Reply:
It has something to do with shared library unloading (destructors). Here we provide a dummy, to keep the compiler happy.

Reply:
But `extern FAR void *__dso_handle[] weak_data;` seems to be a completely another story.
@xiaoxiang781216 do you know anything about the nature of `__dso_handle`?